src/jumpstart/saga.ts

```diff
@@ -0,0 +1,152 @@
+import BigNumber from 'bignumber.js'
+import { JumpstartEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { jumpstartLinkHandler } from 'src/jumpstart/jumpstartLinkHandler'
+import {
+  jumpstartClaimFailed,
+  jumpstartClaimStarted,
+  jumpstartClaimSucceeded,
+} from 'src/jumpstart/slice'
+import { NftMetadata } from 'src/nfts/types'
+import { tokensByIdSelector } from 'src/tokens/selectors'
+import { getTokenId } from 'src/tokens/utils'
+import { addStandbyTransaction } from 'src/transactions/actions'
+import { TokenTransactionTypeV2 } from 'src/transactions/types'
+import Logger from 'src/utils/Logger'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { publicClient } from 'src/viem'
+import networkConfig, { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { all, call, fork, put, select } from 'typed-redux-saga'
+import { Hash, TransactionReceipt, parseAbi, parseEventLogs } from 'viem'
+
+const TAG = 'WalletJumpstart'
+
+export function* jumpstartClaim(privateKey: string) {
+  const walletAddress = yield* select(walletAddressSelector)
+  if (!walletAddress) {
+    Logger.error(TAG, 'No wallet address found in store. This should never happen.')
+    return
+  }
+
+  try {
+    ValoraAnalytics.track(JumpstartEvents.jumpstart_started)
+    yield* put(jumpstartClaimStarted())
+
+    const transactionHashes = yield* call(jumpstartLinkHandler, privateKey, walletAddress)
+    yield* fork(dispatchPendingTransactions, transactionHashes)
```

Comment:
I agree that we should handle networks in a generic way and avoid network hardcoding. Thanks for pointing this out!

I ended up looping through all remote config networkIds, but I think it's suboptimal. If the amount of jumpstart-supported networks grows, we would have to loop trough them all, because we have no idea for which specific network that link is. To avoid looping, perhaps we can encode network id within the jumpstart link?

Reply:
Also, we probably want to refactor jumpstartLinkHandler from ContractKit to Viem to make it multichain? 

For know, I just made it throw an error if network id isn't Celo.